ludwig/datasets/kdd_appetency/__init__.py

```diff
@@ -0,0 +1,134 @@
+                   BaseDataset):
+    """
+    The KDD Cup 2009 Appetency dataset
+
+    Additional Details:
+
+    https://www.kdd.org/kdd-cup/view/kdd-cup-2009/Data
+    """
+    def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION, label=None):
+        super().__init__(dataset_name="kdd_appetency", cache_dir=cache_dir)
+
+    def read_file(self, filetype, filename, header=0):
+        if filetype == 'data':
+            file_df = pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=header, sep='\t')
+        elif filetype == 'labels':
+            file_df = -pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=None)[0]
+        elif filetype == 'txt':
+            file_df = pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=None)
+        return file_df
+
+    def process_downloaded_dataset(self, header=0):
+        super().process_downloaded_dataset()
+        unprocessed_df = pd.read_csv(os.path.join(self.processed_dataset_path,
+                                                self.csv_filename))
+        
+        train_df = unprocessed_df[unprocessed_df[SPLIT] == 0]
+        self.test_df = unprocessed_df[unprocessed_df[SPLIT] == 2]
+        
+        categorical_features = { 190, 191, 192, 193, 194, 195, 196, 197, 198,
+                                199, 200, 201, 202, 203, 204, 205, 206, 207,
+                                209, 210, 211, 212, 213, 214, 215, 216, 217,
+                                218, 219, 220, 221, 222, 223, 224, 225, 226,
+                                227, 228 }
+
+        train_df = process_categorical_features(train_df, categorical_features)
+        train_df = process_numerical_features(train_df, categorical_features)
+        
+        target = self.read_file('labels', os.path.join(self.raw_dataset_path,
+                                                       'orange_small_train_appetency.labels'))
+        
+        train_idx = self.read_file('txt', os.path.join(self.raw_dataset_path,
+                                                       'stratified_train_idx_appetency.txt'))
+        
+        val_idx = self.read_file('txt', os.path.join(self.raw_dataset_path,
+                                                       'stratified_test_idx_appetency.txt'))
+        
+        processed_train_df = train_df.iloc[train_idx[0]].copy()
+        processed_train_df['target'] = target.iloc[train_idx[0]]
+        
+        processed_val_df = train_df.iloc[val_idx[0]].copy()
+        processed_val_df['target'] = target.iloc[val_idx[0]]
+        processed_val_df['split'] = 1
+        
+        self.data_df = pd.concat([processed_train_df, processed_val_df])
+
+    def load_processed_dataset(self, split) -> Tuple[pd.DataFrame,
+                                                     pd.DataFrame,
+                                                     pd.DataFrame]:
+        if split:
+            if SPLIT in self.data_df:
```

Comment:
the pattern of setting `self.data_df` can be improved. We want the different steps (download, process and load) to be atomic and independent, because if the directories are already crated, they are not repeated, but this breaks that assumption.